pkgs/development/python-modules/cyclonedds-python/default.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+, cyclonedds
+, setuptools
+, rich-click
+
+, pytest
+, pytest-cov
+, pytest-mock
+, pytest-asyncio
+}:
+
+buildPythonPackage rec {
+  pname = "cyclonedds-python";
+  version = "0.10.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "eclipse-cyclonedds";
+    repo = pname;
+    rev = version;
+    hash = "sha256-MN3Z5gqsD+cr5Awmsia9+uCHL/a2KQP2uMS13rVc1Hw=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  buildInputs = [ cyclonedds ];
+
+  pythonPath = [ rich-click ];
+
+  checkInputs = [
+    pytest
+    pytest-cov
+    pytest-mock
+    pytest-asyncio
+  ];
+
+  env.CYCLONEDDS_HOME = "${cyclonedds.out}";
+
+  pipInstallFlags = [ "--no-binary" ];
+
+  checkPhase = ''
+    python3 -m pytest tests
+  '';
+
+  meta = with lib; {
+    description = "Python binding for Eclipse Cyclone DDS";
+    homepage = "https://github.com/eclipse-cyclonedds/cyclonedds-python";
+    license = licenses.epl20;
+    maintainers = with maintainers; [ kvik ] ++ teams.python.members;
```

```suggestion
    maintainers = with maintainers; [ kvik ];
```

Please maintain the package yourself as good as possible.